packages/beacon-node/src/util/sszBytes.ts

```diff
@@ -68,32 +76,51 @@ export function getAttDataBase64FromAttestationSerialized(data: Uint8Array): Att
  * Extract aggregation bits from attestation serialized bytes.
  * Return null if data is not long enough to extract aggregation bits.
  */
-export function getAggregationBitsFromAttestationSerialized(data: Uint8Array): BitArray | null {
-  if (data.length < VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE + SIGNATURE_SIZE) {
+export function getAggregationBitsFromAttestationSerialized(fork: ForkName, data: Uint8Array): BitArray | null {
+  const aggregationBitsStartIndex =
+    ForkSeq[fork] >= ForkSeq.electra
+      ? VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE + COMMITTEE_BITS_SIZE + SIGNATURE_SIZE
+      : VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE + SIGNATURE_SIZE;
+
+  if (data.length < aggregationBitsStartIndex) {
     return null;
   }
 
-  const {uint8Array, bitLen} = deserializeUint8ArrayBitListFromBytes(
-    data,
-    VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE + SIGNATURE_SIZE,
-    data.length
-  );
+  const {uint8Array, bitLen} = deserializeUint8ArrayBitListFromBytes(data, aggregationBitsStartIndex, data.length);
   return new BitArray(uint8Array, bitLen);
 }
 
 /**
  * Extract signature from attestation serialized bytes.
  * Return null if data is not long enough to extract signature.
  */
-export function getSignatureFromAttestationSerialized(data: Uint8Array): BLSSignature | null {
-  if (data.length < VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE + SIGNATURE_SIZE) {
+export function getSignatureFromAttestationSerialized(fork: ForkName, data: Uint8Array): BLSSignature | null {
+  const signatureStartIndex =
+    ForkSeq[fork] >= ForkSeq.electra
+      ? VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE + COMMITTEE_BITS_SIZE
+      : VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE;
+
+  if (data.length < signatureStartIndex + SIGNATURE_SIZE) {
     return null;
   }
 
-  return data.subarray(
-    VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE,
-    VARIABLE_FIELD_OFFSET + ATTESTATION_DATA_SIZE + SIGNATURE_SIZE
-  );
+  return data.subarray(signatureStartIndex, signatureStartIndex + SIGNATURE_SIZE);
+}
+
+/**
+ * Extract committee bits from Electra attestation serialized bytes.
+ * Return null if data is not long enough to extract committee bits.
+ */
+export function getCommitteeBitsFromAttestationSerialized(data: Uint8Array): BitArray | null {
```

Comment:
should return `BitVector`?
also add a unit test for this

Reply:
> should return `BitVector`? 

Hmmm I think `BitVector` is an overkill. The caller to this function only requires a bit array that can conveniently retrieve `getSingleTrueBit()`. 

Perhaps the same reasoning for `getAggregationBitsFromAttestationSerialized()` returning `BitArray` and not `BitList`. 

> also add a unit test for this

added